pkgs/development/python-modules/bootstrapped-build/default.nix

```diff
@@ -0,0 +1,65 @@
+{ stdenv
+, python
+#, buildBuildHook
+, pipInstallHook
+, build
+, wheel
+, pip # Should be replaced by `installer`
+, requiredPythonModules
+, unzip
+, setuptools
+, flit-core
+, wrapPython
+}:
+
+let
+  runtimeDeps = requiredPythonModules [ build wheel ];
+  srcs = with stdenv.lib; remove python.src (catAttrs "src" runtimeDeps);
+in stdenv.mkDerivation rec {
+  pname = "build";
+  inherit (build) version;
+  name = "${python.libPrefix}-bootstrapped-${pname}-${version}";
+
+  inherit srcs;
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    unzip
+    wrapPython
+    pip # Replace with `installer`
+    # Should collect the build backends
+    setuptools
+    python
+    wheel
+    flit-core
+
+  ];
+
+  buildInputs = [
+    python
+  ];
+
+  propagatedBuildInputs = [
+    #(buildInstallHook.override({build=null;}))
+    #pipInstallHook
+  ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    export PYTHONPATH="$(ls | tr '\n' '\:'):''${PYTHONPATH:-}"
+    for pkg in $(find . -mindepth 1 -maxdepth 1 -type d -exec basename {} \;); do
```

Comment:
for every runtime dep of `build` we create a wheel and install it. This requires the build backends to be available already. We probably end up needing to add setuptools and flit here as well.